### DISCUSSION TITLE
Tool to extract JSON field+values from rtl_433.json into CSV format

### DISCUSSION THREAD

**Discussion by hdtodd** [merbanan/rtl_433#2451] [Show and tell]
I found it helpful to extract specific fields and associated values from rtl_433 JSON log files into CSV format for subsequent processing.  I tired of writing one-off Python scripts to extract various fields for different analyses, so I wrote a more generalized tool.  

In case it's useful, it's posted in repository [rtl_433_stats](https://github.com/hdtodd/rtl_433.stats) under py-version/tools as `xtract-json`.  Instructions for its use can be obtained by appending the '-h' switch.

For example, working from the `rtl_433_stats` directory, the command line:
`py-version/tools/xtract-json -i xaa.json -f battery_ok status -o x; sort x | uniq `
produces the following report:
```
0Device_key,                            battery_ok,    status,
Acurite-01185M/0/0,                              1,          ,
Acurite-606TX//134,                              1,          ,
Acurite-609TXC//194,                             0,        10,
Acurite-Tower/A/11524,                           1,          ,
LaCrosse-TX141Bv3/1/253,                         1,          ,
LaCrosse-TX141THBv2/0/168,                       1,          ,
Prologue-TH/2/203,                               0,          ,
```
(The '0' before "Device" is intentional to allow for processing by sort)

I'm currently exploring battery and status changes over extended periods of time, so over a 2-month period, that report tells me what to look for as changes in the "battery_ok" and "status" fields for various devices.  

Before writing this program, I did a similar extract to study the inter-transmission gap times and packet-per-transmission counts, so this generalized tool might be useful for other analyses of device characteristics over long terms.

**Comment by zuckschwerdt** [comment]
You might find [`jq`](https://stedolan.github.io/jq/) interesting.

**Comment by hdtodd** [comment]
Christian,

Yes, thanks!  I'm expecting to need something like  jq, so I've 
downloaded and tested it and will keep it for future use.  Right now, 
doing a lot of analysis for which a CSV format is better suited, but I 
expect to need to do some JSON processing, too.
